### PR TITLE
chore/ingress-additions

### PR DESCRIPTION
**Pull request by mosheavni**
fix nil on `hosts`
fix ingress class, from annotation or from class property
open an ingress link in the browser

### CODE REVIEW

File: ftplugin/k8s_ingresses.lua [hunk 1]
```diff
@@ -13,6 +14,58 @@ local function set_keymap(bufnr)
       overview_view.View()
     end,
   })
+  api.nvim_buf_set_keymap(bufnr, "n", "gx", "", {
```
Comment (Ramilito) [inline ftplugin/k8s_ingresses.lua:17]: I love this!

**Comment by mosheavni** [comment]
the keymap doesn't appear on hints. mind stepping in here?

File: ftplugin/k8s_ingresses.lua (hunk 1 above)
Comment (Ramilito) [reply ftplugin/k8s_ingresses.lua:17]: We need to add a plug function like this:
```
api.nvim_buf_set_keymap(bufnr, "n", "gx", "<Plug>(kubectl.browse)", {
```

File: lua/kubectl/views/ingresses/definition.lua [hunk 2]
```diff
@@ -5,6 +5,9 @@ local M = {
   display_name = "Ingresses",
   ft = "k8s_ingresses",
   url = { "{{BASE}}/apis/networking.k8s.io/v1/{{NAMESPACE}}ingresses?pretty=false" },
+  hints = {
+    { key = "<Plug>(kubectl.browse)", desc = "browse", long_desc = "Open host in browser" },
```
Comment (Ramilito) [inline lua/kubectl/views/ingresses/definition.lua:9]: and also add a hint here ☝️

File: ftplugin/k8s_ingresses.lua [hunk 3]
```diff
@@ -1,7 +1,11 @@
 local api = vim.api
+local ResourceBuilder = require("kubectl.resourcebuilder")
 local ingresses_view = require("kubectl.views.ingresses")
 local loop = require("kubectl.utils.loop")
 local overview_view = require("kubectl.views.overview")
+local mappings = require("kubectl.mappings")
+
+mappings.map_if_plug_not_set("n", "gx", "<Plug>(kubectl.browse)")
```
Comment (Ramilito) [inline ftplugin/k8s_ingresses.lua:8]: And then the actual mapping here ☝️

**Comment by Ramilito** [comment]
> the keymap doesn't appear on hints. mind stepping in here?

I did some ninja commits, but haven't verified that they actually work

**Comment by mosheavni** [comment]
image
any idea why?

**Comment by mosheavni** [comment]
oh this stupid `vim.notify()`. i removed it

**Comment by mosheavni** [comment]
ready!

**Comment by Ramilito** [comment]
Awesome idea about the browse to host btw 👏

**Pull request merged**